src/views/Gas.js

```diff
@@ -0,0 +1,395 @@
+import React, { useState, useEffect } from 'react'
+import { withRouter } from 'react-router-dom'
+import Chart from 'chart.js';
+import $ from 'jquery';
+
+// number of consecutive block we take the average from 
+const averageLength = 3
+const blockLimit = 5* 10**9
+
+function Gas({client,head}) {
+    const [stats, setStat] = useState(new Stats(client, averageLength))
+
+    const updateStats = async () => {
+        const s = stats
+        await s.fetchCids()
+        setStat(s)
+    }
+
+    useEffect( () =>  {
+        updateStats()
+    },[client,head])
+
+    return (
+        <div className="row">
+            <div className="accordion col-12" id="accordionExample">
+                    <div className="card">
+                    <div className="card-header" id="biggestGas">
+                      <h2>
+                        <button className="btn btn-link" type="button" data-toggle="collapse" data-target="#collapseBiggest" aria-expanded="true" aria-controls="collapseBiggest">
+                            Biggest gas users
+                        </button>
+                      </h2>
+                    </div>
+                    <div id="collapseBiggest" className="collapse" aria-labelledby="biggestGas" data-parent="#accordionExample">
+                        <div className="card-body">
+                            <BiggestGasSpender nstats={stats} head={head} />
+                        </div>
+                    </div>
+                </div>
+                <div className="card">
+                    <div className="card-header" id="headingOne">
+                      <h2>
+                        <button className="btn btn-link" type="button" data-toggle="collapse" data-target="#collapseOne" aria-expanded="false" aria-controls="collapseOne">
+                            Gas usage table
+                        </button>
+                      </h2>
+                    </div>
+                    <div id="collapseOne" className="collapse" aria-labelledby="headingOne" data-parent="#accordionExample">
+                        <div className="card-body">
+                            <GasTable nstats={stats} head={head} />
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    )
+}
+
+function BiggestGasSpender({head,nstats}) {
+    const [dataset,setData] = useState({})
+    const [opts,setOpts] = useState({})
+    const [pie,setPie] = useState(undefined)
+    const maxUser = 10
+    const id = "pieBiggestGas"
+
+    const setCollapseEvent = pie => {
+        // XXX This is bad because it requires names of ID to be constant but 
+        // it'll do for now
+        $("#collapseBiggest").on('show.bs.collapse', function () {
+            pie.update();
+        });
+        $("#collapseBiggest").on('hide.bs.collapse', function () {
+            pie.update();
+        });
+    }
```

Comment:
is this jquery? not sure this is the right design pattern, we should fix this (things may not work as expected)